stdlib/camlinternalMod.ml

```diff
@@ -28,6 +28,38 @@ let overwrite o n =
     Obj.set_field o i (Obj.field n i)
   done
 
+let overwrite_closure o n =
+  (* We need to use the [raw_field] functions at least on the code
+     pointer, which is not a valid value in -no-naked-pointers
+     mode. *)
+  assert (Obj.tag n = Obj.closure_tag);
+  assert (Obj.size o >= Obj.size n);
+  let n_start_env = Obj.Closure.((info n).start_env) in
+  let o_start_env = Obj.Closure.((info o).start_env) in
+  (* if the environment of n starts before the one of o,
+     clear the raw fields in between. *)
+  for i = n_start_env to o_start_env - 1 do
+    Obj.set_raw_field o i Nativeint.one
+  done;
+  (* if the environment of o starts before the one of n,
+     clear the environment fields in between. *)
+  for i = o_start_env to n_start_env - 1 do
+    Obj.set_field o i (Obj.repr ())
+  done;
+  for i = 0 to n_start_env - 1 do
+    (* code pointers, closure info fields, infix headers *)
+    Obj.set_raw_field o i (Obj.raw_field n i)
+  done;
+  for i = n_start_env to Obj.size n - 1 do
+    (* environment fields *)
+    Obj.set_field o i (Obj.field n i)
+  done;
+  for i = Obj.size n to Obj.size o - 1 do
+    (* clear the leftover space *)
+    Obj.set_field o i (Obj.repr ())
+  done;
```

Comment:
This is not strictly necessary.  In this particular use case, I think those fields are already `()` /  integer 0.  In general, they are well formed values and can stay there.

Reply:
I think that it's slightly nicer to have them, for example it lets us not worry about liveness extensions of previous field values. I don't think this would be an issue with the restricted use-cases of `updateMod`, but it's nice to not have to think about it at all. (But: I added it on a whim and can remove it if you prefer.)

Reply:
OK, your choice.